testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1584,6 +1585,43 @@ private Optional<ZonedDateTime> getMaterializedViewLastFreshTime(String material
         return Optional.ofNullable(lastFreshTime);
     }
 
+    @Test
+    public void testColumnCommentMaterializedView()
+    {
+        if (!hasBehavior(SUPPORTS_COMMENT_ON_MATERIALIZED_VIEW)) {
+            if (hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
+                String viewName = "test_materialized_view_" + randomNameSuffix();
+                assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM nation");
+                assertQueryFails("COMMENT ON COLUMN " + viewName + ".regionkey IS 'new region key comment'", "This connector does not support setting materialized view column comments");
+                assertUpdate("DROP MATERIALIZED VIEW " + viewName);
+                return;
+            }
+            throw new SkipException("Skipping as connector does not support MATERIALIZED VIEW COLUMN COMMENT");
+        }
+
+        String catalog = getSession().getCatalog().orElseThrow();
+        String schema = getSession().getSchema().orElseThrow();
+
+        QualifiedObjectName view = new QualifiedObjectName(catalog, schema, "test_materialized_view_" + randomNameSuffix());
+
+        createTestingMaterializedView(view, Optional.empty());
+
+        assertUpdate("COMMENT ON COLUMN " + view.getObjectName() + ".name IS 'new comment'");
+        assertThat(getColumnComment(view.getObjectName(), "name")).isEqualTo("new comment");
+
+        // comment deleted
+        assertUpdate("COMMENT ON COLUMN " + view.getObjectName() + ".name IS NULL");
+        assertThat(getColumnComment(view.getObjectName(), "name")).isEqualTo(null);
+
+        // comment set to non-empty value before verifying setting empty comment
+        assertUpdate("COMMENT ON COLUMN " + view.getObjectName() + ".name IS 'updated comment'");
+        assertThat(getColumnComment(view.getObjectName(), "name")).isEqualTo("updated comment");
+
+        // comment set to empty or deleted
+        assertUpdate("COMMENT ON COLUMN " + view.getObjectName() + ".name IS ''");
+        assertThat(getColumnComment(view.getObjectName(), "name")).isEmpty();
+    }
```

Comment:
Please drop the materialized view in `finally` block. This test may run on shared services.

Reply:
refactored